jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -3874,16 +3878,60 @@ static void checkBoundsBeginEnd(int begin, int end, int length) {
 		throw newStringIndexOutOfBoundsException(begin, end, length);
 	}
 	
+	class StringSpliteratorOfInt implements Spliterator.OfInt {
+		int start; // index of first character in the array
+		int end; // index after the final character
+		int cursor; // next character to read
+		IntFunction func;
+		
+		StringSpliteratorOfInt(int start, int end, IntFunction func) {
+			this.start = start;
+			this.end = end;
+			this.func = func;
+			cursor = start;
+		}
+
+		@Override
+		public long estimateSize() {
+			return end - start;
+		}
+
+		@Override
+		public int characteristics() {
+			return IMMUTABLE | ORDERED | SIZED | SUBSIZED;
+		}
+
+		@Override
+		public OfInt trySplit() {
+			int newEnd = (cursor + end + 1)/2;
```

Comment:
This may overflow for large strings, `cursor + ((end - cursor + 1) / 2)` will not.